src/core/instances.jl

```diff
@@ -351,14 +351,20 @@ function Base.run(mi::ModelInstance, ntimesteps::Int=typemax(Int),
 
     clock = Clock(time_keys)
 
-    # Get the dimensions dictionary
-    dim_val_dict = DimValueDict(dim_dict(mi.md), clock)
+    # Get the dimensions Named Tuple from the dimension dictionary which will be 
+    # passed to run_timestep() and init(), so we can safely implement Base.getproperty(),
+    # allowing `d.regions` etc.
+    # All values in the named tuple are vectors of Ints, except the `:time` value, which is a
+    # vector of AbstractTimesteps, so that `d.time` returns values that can be used for indexing
+    # into timestep arrays.
+    d = Dict([name => (name == :time ? timesteps(clock) : collect(values(dim))) for (name, dim) in dim_dict])
```

Comment:
@davidanthoff this is obviously not necessary, going from a `Dict` to a `NamedTuple`, but I was having a little trouble getting list comprehension to work properly going straight to a `NamedTuple`, any fancy ideas? I can play with it or just allocate and then iteratively append or push.

Reply:
Nevermind got it